internal/tsoptions/parsinghelpers.go

```diff
@@ -510,6 +511,63 @@ func mergeCompilerOptions(targetOptions, sourceOptions *core.CompilerOptions) *c
 	return targetOptions
 }
 
+// mergeCompilerOptionsWithRaw merges the source compiler options into the target compiler options
+// with awareness of explicitly set null values in the raw JSON.
+// Fields in the source options will overwrite the corresponding fields in the target options,
+// including when they are explicitly set to null in the raw configuration.
+func mergeCompilerOptionsWithRaw(targetOptions, sourceOptions *core.CompilerOptions, rawSource any) *core.CompilerOptions {
+	if sourceOptions == nil {
+		return targetOptions
+	}
+
+	// First do the normal merge
+	mergeCompilerOptions(targetOptions, sourceOptions)
+
+	// Then handle explicitly set null values using the raw JSON
+	if rawSource != nil {
+		if rawMap, ok := rawSource.(*collections.OrderedMap[string, any]); ok {
+			if compilerOptionsRaw, exists := rawMap.Get("compilerOptions"); exists {
+				if compilerOptionsMap, ok := compilerOptionsRaw.(*collections.OrderedMap[string, any]); ok {
+					handleExplicitNullValues(targetOptions, compilerOptionsMap)
+				}
+			}
+		}
+	}
+
+	return targetOptions
+}
+
+// handleExplicitNullValues processes explicitly set null values in the raw JSON
+// and applies them to the target options to override inherited values
+func handleExplicitNullValues(targetOptions *core.CompilerOptions, rawCompilerOptions *collections.OrderedMap[string, any]) {
```

Comment:
This function is way too nested; use a more Go-like style with early breaks and continues